### ISSUE TITLE
enhancement: support payment URIs

### ISSUE THREAD

**Issue by gmale** [zcash/zcash-android-wallet#189]
As a user, I'd like to be able to scan Payment URI's per [BIP-0021](https://en.bitcoin.it/wiki/BIP_0021) so that I can transmit payment to others who use these QR codes easily.

Requested by: @chilebob

**Comment by gmale** [comment]
Turns out, we shouldn't use BIP-0021 and instead should implement [ZIP-321](https://github.com/zcash/zips/issues/347). Which, for example, would include better memo support.

**Comment by gmale** [comment]
Blocked on agreement and merging of ZIP-321

**Comment by nathan-at-least** [comment]
It looks like ZIP-321 was merged in the "Proposed" state. Is that enough to unblock this feature in the wallet?

**Comment by nighthawk24** [comment]
Let's reopen this issue or create a new one to add support for ZIP-321

**Comment by gmale** [comment]
Good idea. We should bump this in priority. Especially given that URIs will probably become more important in a world with multiple-address types.

The underlying Rust code is complete. We just need to expose it into the SDK, which involves figuring out what data structure(s) to send over the JNI boundary. @nuttycom and I have been discussing ways to exchange data from `kotlin -> rust` and `swift -> rust` such that it's written in one place in and used everywhere (potentially even WASM). ZIP-321 would be an excellent use case for proving those concepts.

**Comment by nuttycom** [comment]
The reference implementation is at https://github.com/zcash/librustzcash/blob/master/zcash_client_backend/src/zip321.rs
